packages/java-parser/src/productions/blocks-and-statements.js

```diff
@@ -28,12 +28,16 @@ function defineRules($, t) {
     const isLocalVariableDeclaration = this.BACKTRACK_LOOKAHEAD(
       $.isLocalVariableDeclaration
     );
+
+    const isClassDeclaration = this.BACKTRACK_LOOKAHEAD($.isClassDeclaration);
+
     $.OR([
       {
         GATE: () => isLocalVariableDeclaration,
         ALT: () => $.SUBRULE($.localVariableDeclarationStatement)
       },
       {
+        GATE: () => isClassDeclaration,
```

Comment:
Not sure if this is needed ? We got otherwise:
```
 Ambiguous Alternatives Detected: <2 ,3> in <OR> inside <blockStatement> Rule,
<'@', Identifier> may appears as a prefix path in all these alternatives.
```

because a statement can have annotations

Reply:
makes sense